### PR TITLE
external code integration

### PR DESCRIPTION
**Pull request by ivanmaione**
This PR propose a "simple" architecture to run external codes. A test is made applying the proposed architecture for plasmod. This work revisits the work done in #251.

The same architecture should be used for JETTO and PROCESS.

This is still a draft version. Out of the goal of this PR is the interaction/loop of the different codes inside the system-code workflow. This code is only intended to perform the action of:

1. setup all the variable necessary to run the external code
2. run the code in different runmodes
3. implement the necessary functions to get all the data that could be necessary within the system-code workflow.

### CODE REVIEW

File: examples/codes/run_plasmod_example.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+new_params = {
+    "A": 3.1,
+    "B_0": 5.3,
+    "R_0": 8.93,
+    "q_95": 3.23,
+}
+build_config = {
+    "problem settings": {
+        "Pfus_req": 2000,
+        "i_modeltype": 111,
```
[the remark this answers is not in the document]
Comment (je-cook) [reply examples/codes/run_plasmod_example.py]: Now you can specify either

**Comment by je-cook** [comment]
The only thing I think I havent done @DanShort12 is the fortranformat stuff. Will do that tomorrow

**Comment by je-cook** [comment]
Ok @CoronelBuendia @DanShort12 any more thoughts or comments?

**Review by DanShort12 — APPROVED** [review]
Fine with me. There are a couple of bits flagged by SonarCloud but we're not gating on those yet so happy to get this in. We probably need to do some refactoring to reduce complexity in a bunch of places anyway. Happy to get this in assuming the pipeline passes.

File: bluemira/codes/plasmod/mapping.py [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+    # [-] input H-factor:if i_modeltype > 1 H factor calculated
+    # ##### "BM_INP": "hfact_inp",
+    # [-] H-factor
+    # ##### "BM_OUT": "H",  # H_star?
+}
+
+
+def create_mapping():
+    """
+    Creates mappings for plasmod
```
[the remark this answers is not in the document]
Comment (DanShort12) [reply bluemira/codes/plasmod/mapping.py:393]: The name is fine for me as is, was just checking on the behaviour here.

**Pull request merged**